### PR TITLE
e2e_network:stop using deprecated framework.ExpectError

### PR DESCRIPTION
**Pull request by shijinye**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
test/e2e/network stop using deprecated framework.ExpectError

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/pull/115961

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/network/netpol/network_policy_api.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ import (
 	admissionapi "k8s.io/pod-security-admission/api"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
```
Comment (carlory) [inline test/e2e/network/netpol/network_policy_api.go:31]: please leave a blank line

**Comment by carlory** [comment]
@shijinye one nit, otherwise LGTM.

/ok-to-test

**Comment by danwinship** [comment]
/approve
thanks!

**Pull request merged**